packages/components/src/components/hds/reveal/index.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { guidFor } from '@ember/object/internals';
+import { assert } from '@ember/debug';
+
+import type { HdsDisclosurePrimitiveSignature } from '../disclosure-primitive';
+import type { HdsRevealToggleButtonSignature } from './toggle/button';
+
+interface HdsRevealSignature {
+  Args: {
+    text: HdsRevealToggleButtonSignature['Args']['text'];
+    textWhenOpen?: HdsRevealToggleButtonSignature['Args']['text'];
+    isOpen?: HdsRevealToggleButtonSignature['Args']['isOpen'];
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HdsDisclosurePrimitiveSignature['Element'];
+}
+
+export default class HdsRevealComponent extends Component<HdsRevealSignature> {
+  /**
+   * Generates a unique ID for the Content
+   *
+   * @param contentId
+   */
+  contentId = 'content-' + guidFor(this);
+
+  /**
+   * @param getText
+   * @type {string}
+   * @description A local function that emulates a getter to compute the value of the `@text` argument for the button (mainly to make TypeScript happy)
+   */
+  getText = (
+    isOpen: boolean,
+    text: string,
+    textWhenOpen: string | undefined
+  ): string => {
+    if (isOpen && textWhenOpen !== undefined) {
+      return textWhenOpen;
+    } else {
+      if (text !== undefined) {
+        return text;
+      } else {
+        assert('@text for "Hds::Reveal" must have a valid value');
```

Comment:
@alex-ju when do you think we should start to remove these kind of assertions, used as guards against the required arguments, for example, or arguments that should be one of a list? is this covered by TypeScript now (in fact TS is not complaining that I'm not returning a value at line 49/50) or there may be consumers that still benefit from these assertions?

Reply:
yes, I was just reviewing the code – `text` can't be undefined (as per signature; not optional) and it should throw a type error in this case, making the assert unnecessary for TypeScript users, but it would still benefit non-TypeScript consumers, so I believe it's too early to remove them.

Reply:
@alex-ju can you add this specific point (keep the existing assertions in place, because this will benefit non-TypeScript consumers) to [the documentation](https://github.com/hashicorp/design-system/pull/2087)? thanks